### PR TITLE
KS-205: add workflow name to spec

### PR DESCRIPTION
**Pull request by krehermann**
Adds WorkflowName to the spec

This in turn is used on the node to ensure that (owner, name) are unique, which is important for job management.

### CODE REVIEW

File: core/services/job/orm.go [hunk 1]
```diff
@@ -1043,6 +1045,26 @@ func (o *orm) FindJobIDsWithBridge(ctx context.Context, name string) (jids []int
 	return
 }
 
+func (o *orm) FindJobIDByWorkflow(ctx context.Context, spec WorkflowSpec) (jobID int32, err error) {
+	// NOTE: We want to explicitly match on NULL feed_id hence usage of `IS
```
Comment (bolekk) [inline core/services/job/orm.go]: unrelated leftover comment?

**Review by bolekk — COMMENTED** [review]
Do you want to clean up commented out code?

Also several tests are failing.

**Review by ettec — DISMISSED** [review]

File: core/services/job/models.go [hunk 2]
```diff
@@ -859,9 +860,18 @@ func (w *WorkflowSpec) Validate() error {
 		return fmt.Errorf("incorrect length for id %s: expected %d, got %d", w.WorkflowID, workflowIDLen, len(w.WorkflowID))
 	}
 
+	// special case empty owner and name for anonymous workflows
+	if w.WorkflowOwner == "" && w.WorkflowName == "" {
+		return nil
+	}
+
```
Comment (DeividasK) [inline core/services/job/models.go]: If we include this, then there should be test cases, which I can't find.

Comment (krehermann) [reply core/services/job/models.go]: good catch, this was a bad push that was half baked btw our call and slack update. removed.

File: core/services/job/orm.go [hunk 3]
```diff
@@ -78,6 +78,8 @@ type ORM interface {
 
 	DataSource() sqlutil.DataSource
 	WithDataSource(source sqlutil.DataSource) ORM
+
+	FindJobIDByWorkflow(ctx context.Context, spec WorkflowSpec) (int32, error)
```
Comment (bolekk) [inline core/services/job/orm.go:82]: nit: a fully-descriptive name would be FindJobByWorkflowNameAndOwner but it's annoyingly long.

**Review by bolekk — DISMISSED** [review]

**Review by bolekk — DISMISSED** [review]

**Review by bolekk — APPROVED** [review]

**Review by DeividasK — APPROVED** [review]

**Pull request merged**